### PR TITLE
JDK-8273979: move some os time related functions to os_posix for POSIX platforms

### PR DESCRIPTION
**Pull request by MBaesken**
There are a couple of time related functions in os_<platform> for Linux, BSD+Mac and AIX that are pretty similar (or identical) across the platforms and can be centralized in os_posix .
While doing so, I noticed that  os::supports_vtime()   returns always true , on Posix and also on Windows, so we might remove it completely if this prefered ? 

Best regards , Matthias

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change must be properly reviewed

### Issue
 * [JDK-8273979](https://bugs.openjdk.java.net/browse/JDK-8273979): move some os time related functions to os_posix for POSIX platforms

### Reviewers
 * [David Holmes](https://openjdk.java.net/census#dholmes) (@dholmes-ora - **Reviewer**)
 * [Lutz Schmidt](https://openjdk.java.net/census#lucy) (@RealLucy - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.java.net/jdk pull/5582/head:pull/5582` \
`$ git checkout pull/5582`

Update a local copy of the PR: \
`$ git checkout pull/5582` \
`$ git pull https://git.openjdk.java.net/jdk pull/5582/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 5582`

View PR using the GUI difftool: \
`$ git pr show -t 5582`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.java.net/jdk/pull/5582.diff">https://git.openjdk.java.net/jdk/pull/5582.diff</a>

</details>

### CODE REVIEW

File: src/hotspot/os/bsd/os_bsd.cpp [hunk 1]
```diff
@@ -804,52 +784,8 @@ jlong os::javaTimeNanos() {
   // See https://blogs.oracle.com/dave/entry/cas_and_cache_trivia_invalidate
   return (prev == obsv) ? now : obsv;
 }
-
-void os::javaTimeNanos_info(jvmtiTimerInfo *info_ptr) {
```
Comment (dholmes-ora) [inline src/hotspot/os/bsd/os_bsd.cpp:808]: Please restore as per previous comment.

File: src/hotspot/os/posix/os_posix.cpp [hunk 2]
```diff
@@ -87,6 +89,14 @@
 #define assert_with_errno(cond, msg)    check_with_errno(assert, cond, msg)
 #define guarantee_with_errno(cond, msg) check_with_errno(guarantee, cond, msg)
 
+static jlong initial_time_count = 0;
+
+#if defined(_ALLBSD_SOURCE)
+static int clock_tics_per_sec = CLK_TCK;
+#else
+static int clock_tics_per_sec = 100;
+#endif
```
Comment (dholmes-ora) [inline src/hotspot/os/posix/os_posix.cpp:98]: Can't we just defer the conditional initialization to the init() function where we already have to conditionally initialise.

**Review by dholmes-ora — CHANGES_REQUESTED** [review]
Hi Matthias,

Generally this refactoring looks good, though I have a request to restore some of the code - see comments below. Also could you clarify where functions were only "pretty similar" rather than identical as we don't want to introduce subtle incompatibilities.

I think supports_vtime() could probably be removed in a separate RFE.

Thanks,
David

**Review by dholmes-ora — APPROVED** [review]
Thanks Matthias - updates look good.

David

**Comment by MBaesken** [comment]
Hi David, I brought back os::javaTimeNanos_info  following your suggestion and adjusted the initialization of  'clock_tics_per_sec ' in os_posix.

>Also could you clarify where functions were only "pretty similar" rather than identical as we don't want to introduce subtle incompatibilities
Sure, went through them and this is what I found
os::elapsedTime()      identical except comment
os::elapsed_counter()   identical
os::elapsed_frequency() identical
os::supports_vtime()          identical
os::getTimesSecs         identical
os::local_time_string  identical
os::localtime_pd   identical

>I think supports_vtime() could probably be removed in a separate RFE.

OK, I opened https://bugs.openjdk.java.net/browse/JDK-8274051  .

**Review by RealLucy — CHANGES_REQUESTED** [review]

**Review by RealLucy — APPROVED** [review]
My previous comment about clock_ticks_per_sec is obsolete. Just saw the initialisation in os::Posix::init(void) now is the same as in os::init(). The static init value is in effect only for the short time until os::init() is invoked. 
Looks good to me.

**Comment by MBaesken** [comment]
> My previous comment about clock_ticks_per_sec is obsolete. Just saw the initialisation in os::Posix::init(void) now is the same as in os::init(). The static init value is in effect only for the short time until os::init() is invoked.
> Looks good to me.

Hi Lutz, thanks for the review .
In case of worries about  the short time frame between  posix and linux init functions, we could introduce 'os::clock_ticks_per_sec()'  and  move this to os_linux, os_aix os_bsd  etc. to have the initialization at one single point.
Not sure if this is worth the effort?

Best regards, Matthias

**Comment by RealLucy** [comment]
I would not add this complexity. Just leave it as it is.

**Pull request closed without merging**